src/components/DonationPageComponents/DonateWindow.js

```diff
@@ -0,0 +1,225 @@
+            // head
+            .then(response => {
+                console.log(response)
+                navigate(`/approved`)
+            })
+        }
+        else if(localStorage.length > 0){
+            console.log(localStorage.getItem("AuthHeader"))
+            fetch(`https://localhost:7000/api/user/${localStorage.getItem("userId")}/donation`, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+                "Authorization": `Basic ${base64.encode(`${localStorage.getItem("AuthHeader")}`)}`
+            },
+            body: JSON.stringify(donate)
+            })
+            // head
+            .then(response => {
+                console.log(response)
+                navigate(`/approved`)
+            })
+        }
+
+        
+    }
+
+    function cc_format_number(value) {
+        const v = value
+        // Remove all non-digit characters
+            // this first replace is to remove all spaces
+            .replace(/\s+/g, "")
+            // this second replace is to remove all non-digit characters
+            .replace(/\D/g, "")
+            // this third replace is to limit the length to 16 characters
+            .substr(0, 16);
+        const parts = [];
+
+        // Split into 4 character chunks
+        for (let i = 0; i < v.length; i += 4) {
+            parts.push(v.substr(i, 4));
+        }
+
+        return parts.length > 1 ? parts.join(" ") : value;
+    }
+
+    function cc_format_expdate(value){
+        const v = value
+        // Remove all non-digit characters
+            .replace(/\s+/g, "")
+            .replace(/\D/g, "")
+            .substr(0, 4);
+        const parts = [];
+
+        // Split into 2 character chunks
+        for (let i = 0; i < v.length; i += 2) {
+            parts.push(v.substr(i, 2));
+        }
+
+        // THis 
+        return parts.length > 1 ? parts.join(" / ") : value;
+    }
+
+
+  return (
+    
+    <div className='bg-gray-100 p-4'>
+        <div className='text-2xl font-bold '>
+            You are supporting {props.title}
+            <div className='text-sm italic font-normal'>You have until {props.plannedEndDate} to support the cause</div>
+        </div>
+        <div className='flex justify-center text-3xl pt-6'>
+            Help make a difference
+        </div>
+        <form onSubmit={handleSubmit} >
+            <div className='flex  p-4 justify-center border-b'>
+                <div className='flex flex-col relative'>
+                    <label className='flex justify-center'>
+                        <div className='absolute top-4 left-3 font-bold text-3xl'>$</div>
+                            <input
+                                className='border-2 border-gray-300 p-4 rounded-lg text-right text-3xl w-48' 
+                                type="number"
+                                step="0.01"
+                                min="1" 
+                                value={Amount}
```

Comment:
Is there a way to make this input initialize to the placeholder, rather than the initial amount value of "0"? Kind of nitpicky, but the user has to delete the "0" first before they can enter their amount

Reply:
The placeholder is actually just the label. I changed the value to be null so it should have nothing in the input so the user doesn't have to backspace

Reply:
looks good